gldemo/shaders/demo_shader.hpp

```diff
@@ -0,0 +1,26 @@
+#include <GL/gl.h>
+
+#define GLSL_VERSION            "430 core"
+
+static const char* const demo_vertex_shader =
+    "#version " GLSL_VERSION "\n"
+    "layout(location = 0) in vec3 in_position;\n"
+    "layout(location = 1) in vec2 in_uv;\n"
+    
+    "uniform mat4 u_modelview;\n"
+    "uniform mat4 u_projection;\n"
+    "out mediump vec2 uv;\n"
+    "void main() {\n"
+    "    gl_Position = u_projection * u_modelview * vec4(in_position,1.0);\n"
+    "    uv = in_uv;\n"
+    "}\n";
+
+static const char* const demo_fragment_shader =
+    "#version " GLSL_VERSION "\n"
+    "precision mediump float;\n"
+    "uniform highp sampler2D main_tex;\n"
+    "in mediump vec2 uv;\n"
+    "out lowp vec4 outcolor;\n"
+    "void main() {\n"
+    "       outcolor = texture(main_tex, uv);\n"
+    "}\n";
```

Comment:
This is identical to debugview's shader. Perhaps better to have a common shader under `common/`. Of course, we should not do that if we expect the shaders to diverge in the future. Do we anticipate that that would happen?

Reply:
Yes, shaders are intentionally not shared. This is because the debugview and gldemo are only visually identical by "coincidence" right now; that is not necessary and not semantically significant. In the future we may have drastically different things rendering in gldemo and the debugview.